src/linear_algebra/unitary_group.lean

```diff
@@ -0,0 +1,190 @@
+/-
+Copyright (c) 2021 Shing Tak Lam. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Shing Tak Lam
+-/
+import linear_algebra.matrix
+import linear_algebra.nonsingular_inverse
+import data.complex.basic
+
+/-!
+# The Unitrary Group O(n)
+
+This file defines elements of the unitary group `unitary_group n`, also written `U(n)`,
+consisting of all `n` by `n` complex matrices where the conjugate transpose is the inverse.
+In addition, we define the group structure on `unitary_group n` and the embedding into the
+general linear group `general_linear_group ℂ (n → ℂ)`.
+
+## Main Definitions
+
+ * `matrix.unitary_group` is the type of matrices where the transpose is the inverse
+ * `matrix.unitary_group.group` is the group structure (under multiplication)
+ * `matrix.unitary_group.embedding_GL` is the embedding `U(n) → GLₙ(ℝ)`
+
+## References
+
+ * https://en.wikipedia.org/wiki/Unitary_group
+
+## Tags
+
+matrix group, group, unitary group
+
+-/
+
+namespace matrix
+universe u
+open linear_map
+
+section
+
+variables {n m : Type*} [fintype n] [fintype m] [decidable_eq n] [decidable_eq m]
+
+/--
+  The conjugate transpose of a matrix, defined to be a the element-wise conjugate of the transpose.
+-/
+def conj_transpose (M : matrix n m ℂ) : matrix m n ℂ :=
+  M.transpose.map complex.conj
+
+localized "postfix `†`:1500 := conj_transpose" in matrix
```

Comment:
This is the "physicists" notation with the dagger, but I haven't found any superscript `*`s in Unicode. Suggestions are welcome.

Reply:
Nevermind, I've just found the `star_ring` instance for complex matrices.